### DISCUSSION TITLE
Gantt panel

### DISCUSSION THREAD

**Discussion by pprkut** [grafana/grafana#71301] [Dashboards, Panels & Visualizations]
It would be really nice for grafana to have a core panel that supports Gantt-like visualization. Essentially I have data that has a start time and an end time, and I want bars indicating how multiple rows relate to each other on a time scale. Pretty much like the traces panel works for traces, just with table or timeseries data.

There is of course https://grafana.com/grafana/plugins/marcusolsson-gantt-panel/, which provides exactly that, but is marked as unmaintained.
The author suggests switching to https://grafana.com/docs/grafana/latest/visualizations/state-timeline/, but it's not clear to me how that should work. The panel wants to display a state at a certain time, which isn't exactly the same as Gantt. And morphing the gantt data into something the state timeline panel supports seems non-trivial and I couldn't find an example for that anywhere.

There's other users wondering the same: https://community.grafana.com/t/enter-duration-end-time-manually-in-state-timeline/71426

I see multiple options:
- Integrate the Gantt panel in grafana and maintain it there
- Add functionality to the state timeline panel so it supports table/timeseries data with manually defined start/end time

I'm not sure how feasible and in scope the latter is for the state timeline panel, so I think the first option would be the preferable solution.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82603. Feel free to continue the discussion around this there. Thank you!
